src/agentscope/utils/monitor.py

```diff
@@ -326,6 +374,383 @@ def get_metrics(self, filter_regex: Optional[str] = None) -> dict:
+            if not self._exists(cursor, metric_name):
+                return None
+            metric = self._get_metric(cursor, metric_name)
+            return metric["unit"]
+
+    def get_metric(self, metric_name: str) -> Optional[dict]:
+        with sqlite_cursor(self.db_path) as cursor:
+            if not self._exists(cursor, metric_name):
+                return None
+            return self._get_metric(cursor, metric_name)
+
+    def get_metrics(self, filter_regex: Optional[str] = None) -> dict:
+        with sqlite_cursor(self.db_path) as cursor:
+            cursor.execute(f"SELECT * FROM {self.table_name}")
+            rows = cursor.fetchall()
+            metrics = {
+                row[1]: {
+                    "value": row[2],
+                    "quota": row[3],
+                    "unit": row[4],
+                }
+                for row in rows
+            }
+        if filter_regex is None:
+            return metrics
+        else:
+            pattern = re.compile(filter_regex)
+            return {
+                key: value
+                for key, value in metrics.items()
+                if pattern.search(key)
+            }
+
+    def _exists(self, cursor: sqlite3.Cursor, name: str) -> bool:
+        cursor.execute(
+            f"""
+            SELECT 1 FROM {self.table_name}
+            WHERE name = ? LIMIT 1
+        """,
+            (name,),
+        )
+        return cursor.fetchone() is not None
+
+    def exists(self, metric_name: str) -> bool:
+        with sqlite_cursor(self.db_path) as cursor:
+            return self._exists(cursor, metric_name)
+
+    def update(self, values: dict, prefix: Optional[str] = None) -> None:
+        with sqlite_transaction(self.db_path) as cursor:
+            for metric_name, value in values.items():
+                self._add(
+                    cursor,
+                    full_name(
+                        name=metric_name,
+                        prefix=prefix,
+                    ),
+                    value,
+                )
+
+    def _create_update_cost_trigger(
+        self,
+        token_metric: str,
+        cost_metric: str,
+        unit_price: float,
+    ) -> None:
+        with sqlite_transaction(self.db_path) as cursor:
+            cursor.execute(
+                f"""
+                CREATE TRIGGER IF NOT EXISTS
+                "{self.table_name}_{token_metric}_{cost_metric}_price"
+                AFTER UPDATE OF value ON "{self.table_name}"
+                FOR EACH ROW
+                WHEN NEW.name = "{token_metric}"
+                BEGIN
+                    UPDATE {self.table_name}
+                    SET value = value + (NEW.value - OLD.value) * {unit_price}
+                    WHERE name = "{cost_metric}";
+                END;
+                """,
+            )
+
+    def register_budget(
+        self,
+        model_name: str,
+        value: float,
+        prefix: Optional[str] = None,
+    ) -> bool:
```

Comment:
Please add some docstring here

Reply:
this method uses the docstring of `MonitorBase.register_budget` to avoid duplication